### ISSUE TITLE
Support easy loading of textures from an asset catalog

### ISSUE THREAD

**Issue by JohnSundell** [JohnSundell/ImagineEngine#150]
Since Imagine Engine doesn't rely on the default system image loading mechanism (using `UIImage(named:)` on iOS) loading textures from an asset catalog is not super easy. While *possible* it requires you to create `Animation` or `Texture` instances using `Image`, like this:

```swift
let image = Image(named: "player")!
actor.animation = Animation(image: image)
```

It would be nicer if Imagine Engine could automatically search the app's asset catalog for an image with a given name, so that you could simply use:

```swift
actor.animation = Animation(textureNamed: "player")
```

**Comment by jussi80** [comment]
Looks like we have run out of luck with this one: asset catalog is compiled as Assets.car file. That file is Apple's proprietary format which could be opened with [CUICatalog,](http://developer.limneos.net/index.php?ios=7.0&framework=CoreUI.framework&header=CUICatalog.h) which is private class. If we use private class, it is very likely that apps with this will not be passed through AppStore review process.  🧐

**Comment by JohnSundell** [comment]
@jussi80 I don't think we should attempt to parse the asset catalogs ourselves. There's the [`NSDataAsset`](https://developer.apple.com/documentation/uikit/nsdataasset) class which provides programmatic access to an asset catalog, it should be able to be used to access the data we need to load a texture 🙂

**Comment by jussi80** [comment]
@JohnSundell NSDataAsset seems to read everything else than image data which is rather strange. Taken from NSDataSet file 

> /// Data assets are not in the same class of stored content as images, so you cannot use a data asset to get image data for an image.

I added images and some random json filled files to .xcassets and images can't be opened, but those son files are no problem with the NSDataAsset. 🤔
